src/App.tsx

```diff
@@ -17,18 +78,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onChangeQuery={filterCallBack}
+                onChangeSelect={onChangeSelect}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoading && (
+                <Loader />
+              )}
+
+              {isLoading && !errorMessage && (
+                <TodoList
+                  todos={visibleTodos}
+                  onHandleModal={handleModal}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo
+        && !errorMessage
+        && isLoading && (
```

Comment:
It's weird that we render a modal if `isLoading` is true. Also, better to create a variable when you have if you have too many conditions like in this case